model/flow/service_event.go

```diff
@@ -87,6 +88,13 @@ func (se *ServiceEvent) UnmarshalJSON(b []byte) error {
 			return err
 		}
 		event = commit
+	case ServiceEventVersionBeacon:
```

Comment:
this is a basically a duplicate of the code in convert.go.  maybe just have this call MessageToServiceEvent instead? (same below)

Reply:
hmm, looks like you can't use MessageToServiceEvent directly.  If you refactor the marshal / unmarshal functions into an interface object, then these become identical code.

Reply:
https://github.com/onflow/flow-go/pull/4220